addon/mixins/icon.js

```diff
@@ -0,0 +1,14 @@
+import Ember from 'ember';
+const { observer } = Ember;
+
+export default Ember.Mixin.create({
+
+  iconDidChange: observer('icon', function() {
```

Comment:
Lets remove Icon mixin and put this into marker-layer directly, ideally with a comment to explain why we're writing this observer manually. This is a marker-layer concern, so no need to be afraid of putting it there. :)